metadata-io/src/main/java/com/linkedin/metadata/recommendation/candidatesource/RecentlyEditedSource.java

```diff
@@ -0,0 +1,135 @@
+package com.linkedin.metadata.recommendation.candidatesource;
+
+import com.codahale.metrics.Timer;
+import com.datahub.util.exception.ESQueryException;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.metadata.datahubusage.DataHubUsageEventConstants;
+import com.linkedin.metadata.datahubusage.DataHubUsageEventType;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.entity.EntityUtils;
+import com.linkedin.metadata.recommendation.EntityProfileParams;
+import com.linkedin.metadata.recommendation.RecommendationContent;
+import com.linkedin.metadata.recommendation.RecommendationParams;
+import com.linkedin.metadata.recommendation.RecommendationRenderType;
+import com.linkedin.metadata.recommendation.RecommendationRequestContext;
+import com.linkedin.metadata.recommendation.ScenarioType;
+import com.linkedin.metadata.search.utils.ESUtils;
+import com.linkedin.metadata.utils.elasticsearch.IndexConvention;
+import com.linkedin.metadata.utils.metrics.MetricUtils;
+import io.opentelemetry.extension.annotations.WithSpan;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.BucketOrder;
+import org.elasticsearch.search.aggregations.bucket.terms.ParsedTerms;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class RecentlyEditedSource implements RecommendationSource {
+  private final RestHighLevelClient _searchClient;
+  private final IndexConvention _indexConvention;
+  private final EntityService _entityService;
+
+  private static final String DATAHUB_USAGE_INDEX = "datahub_usage_event";
+  private static final String ENTITY_AGG_NAME = "entity";
+  private static final int MAX_CONTENT = 5;
+
+  @Override
+  public String getTitle() {
+    return "Recently Edited";
+  }
+
+  @Override
+  public String getModuleId() {
+    return "RecentlyEditedEntities";
+  }
+
+  @Override
+  public RecommendationRenderType getRenderType() {
+    return RecommendationRenderType.ENTITY_NAME_LIST;
+  }
+
+  @Override
+  public boolean isEligible(@Nonnull Urn userUrn, @Nonnull RecommendationRequestContext requestContext) {
```

Comment:
This is great @CorentinDuhamel !

Would you mind adding a unit test for this class. I understand that existing sources may or may not have these, but I think going forward we'll be requiring (and backfilling) these. 

Cheers
John

Reply:
That's was not in my plan. I have no idea how to do it but if it's done for the "recently viewed" module, I can take inspiration from it.